### ISSUE TITLE
Emit JSON `tracing` events formatted such that cloud logging services can parse the message

### ISSUE THREAD

**Issue by tgeoghegan** [divviup/janus#383]
When we render `tracing` events to JSON and write them to stdout, we end up with message payloads in Google Cloud Logging that look like this:

```
{
  "insertId": "0tvgkcz4ekvg4vhg",
  "jsonPayload": {
    "span": {
      "name": "run"
    },
    "spans": [
      {
        "name": "run"
      }
    ],
    "line_number": 134,
    "level": "INFO",
    "filename": "janus_server/src/binary_utils/job_driver.rs",
    "fields": {
      "max_acquire_count": 10,
      "message": "Acquiring jobs"
    },
    "target": "janus_server::binary_utils::job_driver",
    "threadId": "ThreadId(1)",
    "timestamp": "2022-08-11T00:15:58.817444Z"
  },
  "resource": {
    "type": "k8s_container",
    "labels": {
      "pod_name": "aggregation-job-driver-7f49cd967f-7wl74",
      "container_name": "janus",
      "location": "us-west2",
      "cluster_name": "janus-interop-cluster",
      "project_id": "janus-interop-2022-bringup",
      "namespace_name": "interop-00"
    }
  },
  "timestamp": "2022-08-11T00:15:58.817814901Z",
  "severity": "INFO",
  "labels": {
    "compute.googleapis.com/resource_name": "gke-janus-interop-cl-janus-interop-on-09e7729e-daxu",
    "k8s-pod/pod-template-hash": "7f49cd967f",
    "k8s-pod/app": "aggregation_job_driver"
  },
  "logName": "projects/janus-interop-2022-bringup/logs/stdout",
  "receiveTimestamp": "2022-08-11T00:16:03.480327510Z"
}
```

The contents of `jsonPayload` are what Janus actually writes out. This is rich with valuable metadata, but the problem is that the actual message is in `jsonPayload.fields.message`, and Google Cloud Logging expects it to be at `jsonPayload.message`. This doesn't break anything, except that you have to dig into individual log entries in the GCL console to see what the message is.

We should see if we can either change the layout of the events or configure GCL to look for the message in the right place. We also should consider how Amazon's logging infra ingests messages and ensure that whatever we do here will also work in the EKS/FluentBit/AWS CloudWach Logs case.

**Comment by tgeoghegan** [comment]
@divergentdave found that we can [flatten log events](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/fmt/format/struct.Json.html#method.flatten_event) but losing all the structure seems unfortunate. We can maybe also mess with the `fluentbit-gke` config, but we should make sure that we use a supported mechanism for modifying that (otherwise maybe GKE will overwrite our configs).

**Comment by divergentdave** [comment]
Here's the documentation for selecting a custom LogEntry field to use as a summary. https://cloud.google.com/logging/docs/view/logs-explorer-interface#add_summary_fields This setting does not stick around, but becomes part of the URL query.

![image](https://user-images.githubusercontent.com/181772/184194958-fa469bd1-6aed-449d-abfc-544156e638bc.png)

**Comment by tgeoghegan** [comment]
[tracing-stackdriver](https://crates.io/crates/tracing-stackdriver) provides a tracing layer that formats JSON events in GCP's expected layout, and does some other nice things like translating tracing `Level` to GCP `LogSeverity`.

**Comment by divergentdave** [comment]
GKE's FluentBit is not configurable, so using FluentBit filters would require changing the cluster settings to "system-only logging" and self-managing a FluentBit DaemonSet. https://cloud.google.com/stackdriver/docs/solutions/gke/managing-logs#custom_agents

**Comment by divergentdave** [comment]
#418 could fix this, but I'm leaving it on the shelf until a conditional compilation fix to tracing-stackdriver makes it to a release.

**Comment by divergentdave** [comment]
Fixed by #418.
